Full Pydantic Model (#324)
stravalib/stravalib: docs/contributing/development-guide.md

$ pip install -e .
```

## Architecture Overview

![Stravalib Architecture](../images/stravalib_architecture.png)

Stravalib contains the following main components:

At the core, a (pydantic) domain model is generated and updated by a bot via pull requests.
This model reflects the officially published API specification by Strava and is stored in

> the module `strava_model.py`. This file should never be edited other than by the pull requests
> generated by the bot.

Reviewer: ```suggestion
the module `strava_model.py`. This file should never be edited  manually. Instead, the stravalib bot will suggest changes to the model through pull requests that can then be merged by stravalib maintainers.
```

> For allowing custom typing, unit conversion, (de-)serialization behavior, and support of
> undocumented Strava features, the module `model.py` contains classes that inherit from the
> official Strava domain model in `strava_model.py`.

Reviewer: ```suggestion
The module `model.py` contains classes that inherit from the
official Strava domain model in `strava_model.py`. This module, supports custom typing, unit conversion, (de-)serialization behavior, and support for
undocumented Strava features, .
```

not sure if this is what you mean just found this sentence a bit confusing as written.

> The module `protocol.py` manages the low-level HTTP traffic (including rate limiting)

Reviewer: what do you mean by low level in this case? could you clarify a bit?
